vts/appraisal/appraisal.go

```diff
@@ -5,27 +5,32 @@ package appraisal
 
 import (
 	"encoding/base64"
+	"fmt"
+	"strings"
 
 	"github.com/veraison/ear"
 	"github.com/veraison/services/config"
+	"github.com/veraison/services/policy"
 	"github.com/veraison/services/proto"
 )
 
 // Appraisal provides an appraisal context internally within the VTS (e.g. for
 // policy evaluation). It is the analog of proto.AppraisalContext, but with a
 // deserialized AttestationResult.
 type Appraisal struct {
+	Scheme          string
 	EvidenceContext *proto.EvidenceContext
 	Result          *ear.AttestationResult
 	SignedEAR       []byte
 }
 
-func New(tenantID string, nonce []byte, submodName string) *Appraisal {
+func New(tenantID string, nonce []byte, scheme string) *Appraisal {
 	appraisal := Appraisal{
+		Scheme: scheme,
 		EvidenceContext: &proto.EvidenceContext{
 			TenantId: tenantID,
 		},
-		Result: ear.NewAttestationResult(submodName, config.Version, config.Developer),
+		Result: ear.NewAttestationResult(scheme, config.Version, config.Developer),
```

Comment:
I have slight problem modifying this interface to EAR.

A scheme is replaced by submodName. 

There could be cases, when a single Scheme can appraise multiple SubMods and so we should retain submodName (as done previously) and add a specific Argument to EAR NewAttestationResult that lets one set the PolicyID for the subMod

Reply:
The interface of `NewAttestationResult` isn't changing here. I'm renaming the argument that is being passed to it. That is not part of the interface. The corresponding parameter name is still `submodName`. 

The parateter of `Appraisal.New` is changing, however that is to more accurately reflect it's current useaged -- it is now used not only to instantiate an attestation result, but also initialize the policy ID (where it is not used as a submod name).

In the context of how `Appraisal.New` is used, it doesn't make sense to pass anything other than the scheme for this parameter, as is done early on, before eveidence handling has begun.

It is not the job of `Appraisal.New` to full initialize the submods of the attestation result. The only reason it needs a submod name at all is because an attestation result is required to have a least one submod, and we need to be able to return an attestation result early on.

The use case you're describing will need to be handled by the scheme-specific `EvidenceHandler` (probably inside `AppraiseEvidence`), where it would rename and/or add additional submods into the attestation result.

A case can be made for adding a new stage to our verification pipeline to be able to query the correct scheme-specific submod names, however that is a significant modification to the current implementation, involving the addition of a new API to the `IEvidenceHandler` interface, and necessitating corresponding changes in the existing plugin implemenations.  This is well outside the scope of the feature being implemented by this pull request.